dispatches/models/fossil_case/concrete_tube/heat_exchanger_tube.py

```diff
@@ -0,0 +1,503 @@
+            None
+        """
+        tube_units = self.config.tube_side.property_package.\
+            get_metadata().get_derived_units
+
+        self.d_tube_outer = Var(initialize=0.011,
+                                doc="Outer diameter of tube",
+                                units=tube_units("length"))
+        self.d_tube_inner = Var(initialize=0.010,
+                                doc="Inner diameter of tube",
+                                units=tube_units("length"))
+
+        self.tube_heat_transfer_coefficient = Var(
+            self.flowsheet().config.time,
+            self.tube.length_domain,
+            initialize=50,
+            doc="Heat transfer coefficient",
+            units=tube_units("heat_transfer_coefficient")
+        )
+
+        self.temperature_wall = Var(
+            self.flowsheet().config.time,
+            self.tube.length_domain,
+            initialize=298.15,
+            units=tube_units("temperature")
+        )
+
+        # Energy transfer between tube wall and tube
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.tube.length_domain,
+            doc="Convective heat transfer",
+        )
+        def tube_heat_transfer_eq(self, t, x):
+            return self.tube.heat[t, x] == self.tube_heat_transfer_coefficient[
+                t, x
+            ] * c.pi * pyunits.convert(self.d_tube_inner,
+                                       to_units=tube_units("length")) * (
+                pyunits.convert(self.temperature_wall[t, x],
+                                to_units=tube_units('temperature')) -
+                self.tube.properties[t, x].temperature
+            )
+
+        # Define tube area in terms of tube diameter
+        self.area_calc_tube = Constraint(
+            expr=4 * self.tube_area == c.pi * pyunits.convert(
+                self.d_tube_inner, to_units=tube_units("length"))**2
+        )
+
+    def initialize(
+        blk,
+        tube_state_args=None,
+        outlvl=idaeslog.NOTSET,
+        solver="ipopt",
+        optarg={"tol": 1e-6},
+    ):
+        """
+        Initialization routine for the unit (default solver ipopt).
+
+        Keyword Arguments:
+            state_args : a dict of arguments to be passed to the property
+                         package(s) to provide an initial state for
+                         initialization (see documentation of the specific
+                         property package) (default = {}).
+            outlvl : sets output level of initialization routine
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+
+        opt = SolverFactory(solver)
```

Comment:
Here you should import `get_solver` from `idaes.core.util` and do `opt = get_solver(solver, optarg)`.